segmentstore/server/src/main/java/io/pravega/segmentstore/server/security/TLSConfigChangeHandler.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.security;
+
+import io.netty.handler.ssl.SslContext;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+@RequiredArgsConstructor
+@Slf4j
+public class TLSConfigChangeHandler {
+
+    /**
+     * A counter representing the number of times this object has been asked to
+     * consume an event.
+     */
+    private final AtomicInteger numOfConfigChangesSinceStart = new AtomicInteger(0);
+
+    private @NonNull final AtomicReference<SslContext> sslContext;
+    private @NonNull final String pathToCertificateFile;
+    private @NonNull final String pathToKeyFile;
+
+    public void handleTlsConfigChange() {
+        log.info("Current reload count = {}", numOfConfigChangesSinceStart.incrementAndGet());
+        sslContext.set(TLSHelper.newServerSslContext(pathToCertificateFile, pathToKeyFile));
+    }
+
+    int getNumOfConfigChangesSinceStart() {
```

Comment:
Should this be an exposed metric?

Reply:
I don't think it'll be useful as a metric. It's just a convenience method for aid in unit testing as of now. Also, the metric is logged, to make it easier to identify when the SSL context is renewed.